frontend/src/Components/StudentMetrics.tsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+import { AxiosError } from 'axios';
+import API from '@/api/api';
+import { Facility, LoginMetrics, ServerResponseOne } from '@/common';
+import {
+    BarChart,
+    CartesianGrid,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+    Bar
+} from 'recharts';
+import PrimaryButton from './PrimaryButton';
+
+const OperationalInsights = () => {
+    const [facilities, setFacilities] = useState<Facility[]>();
+    const [facility, setFacility] = useState('all');
+    const [days, setDays] = useState(7);
+    const [resetCache, setResetCache] = useState(false);
+
+    const { data, error, isLoading, mutate } = useSWR<
+        ServerResponseOne<LoginMetrics>,
+        AxiosError
+    >(
+        `/api/login-metrics?facility=${facility}&days=${days}&reset=${resetCache}`
+    );
+
+    useEffect(() => {
+        void mutate();
+    }, [facility, days, resetCache]);
+
+    useEffect(() => {
+        const fetchFacilities = async () => {
+            const response = await API.get<Facility>('facilities');
+            setFacilities(response.data as Facility[]);
+        };
+        void fetchFacilities();
+    }, []);
+
+    const metrics = data?.data;
+
+    return (
+        <div className="p-8">
+            {error && <div>Error loading data</div>}
+            {!data || (isLoading && <div>Loading...</div>)}
+            {data && metrics && (
+                <>
+                    <PrimaryButton
+                        className="w-30"
+                        onClick={() => setResetCache(!resetCache)}
+                    >
+                        Refresh Data
+                    </PrimaryButton>
```

Comment:
primary button will soon disappear so try to use a just regular button with the class "button" on there